src/main/java/org/broadinstitute/consent/http/service/NihService.java

```diff
@@ -12,23 +13,33 @@
 import java.util.Calendar;
 import java.util.Date;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 public class NihService {
 
     private ResearcherService researcherService;
+    private LibraryCardDAO libraryCardDAO;
 
     @Inject
-    public NihService(ResearcherService researcherService) {
+    public NihService(ResearcherService researcherService, LibraryCardDAO libraryCardDAO) {
         this.researcherService = researcherService;
+        this.libraryCardDAO = libraryCardDAO;
     }
 
-    public List<UserProperty> authenticateNih(NIHUserAccount nihAccount, AuthUser user) throws BadRequestException {
+    public List<UserProperty> authenticateNih(NIHUserAccount nihAccount, AuthUser authUser, Integer userId) throws BadRequestException {
         if (StringUtils.isNotEmpty(nihAccount.getNihUsername()) && !nihAccount.getNihUsername().isEmpty()) {
             nihAccount.setEraExpiration(generateEraExpirationDates());
             nihAccount.setStatus(true);
-            return researcherService.updateProperties(nihAccount.getNihMap(), user, false);
+            List<UserProperty> updatedProps = researcherService.updateProperties(nihAccount.getNihMap(), authUser, false);
+            Optional<UserProperty> eraCommonsProp = updatedProps.stream().filter((prop) -> prop.getPropertyKey() == UserFields.ERA_COMMONS_ID.getValue()).findFirst();
```

Comment:
The commonsID is saved as `nihUsername`, so you'll need to use the `UserFields.ERA_USERNAME` constant
<img width="395" alt="Screen Shot 2021-06-17 at 7 44 31 AM" src="https://user-images.githubusercontent.com/6664778/122390350-e4982e80-cf3f-11eb-9f92-0af7c62296f1.png">

You also need to use `.equals()`, (`==` checks if they're the same reference).

Reply:
@JVThomas  I'm not seeing this. When I look in the user properties table I see usernames saved for that property (nihusername) and I see id numbers saved under the eraCommonsId property. Even in the example you show here it looks like that is meant to be a fake/mock username not ID, where are you seeing IDs saved under nihUsername?

Reply:
This isn't table results, it's the current value of updatedProps at a breakpoint place on line 35. You can mimic the current staging setup by running local instances of the UI and back-end with staging configurations and update the eraCommonsID on the researcherProfile page.

Reply:
It's also visible if you follow the era auth link in the ui, then go to consent swagger and hit the /api/user/me endpoint. You'll see that your user property is saved like this:
```
...
  "researcherProperties": [
    {
      "propertyId": 2578,
      "userId": 3351,
      "propertyKey": "nihUsername",
      "propertyValue": "grushton"
    },
...
```